src/virus_clade_utils/util/sequence.py

```diff
@@ -54,6 +62,74 @@ def get_covid_genome_data(released_since_date: str, base_url: str, filename: str
     logger.info("NCBI API call completed", elapsed=elapsed)
 
 
+def download_covid_genome_metadata(url: str, data_path: Path) -> Path:
```

Comment:
Ported from https://github.com/rogersbw/clade_data_utils/blob/main/update_clades_list.py#L11 with a few modifications:

* put the download in a small function so it can also be used by the target data process
* used the `requests` package for downloading to take advantage of its [retry capabilities ](https://github.com/rogersbw/clade_data_utils/blob/main/update_clades_list.py#L11)